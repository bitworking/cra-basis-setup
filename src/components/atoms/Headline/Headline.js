import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const H1 = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: ${props => (props.primary ? "red" : "gray")};
`;

const Headline = props => <H1 {...props} />;

Headline.propTypes = {
  /** the headline text */
  children: PropTypes.string.isRequired,
  primary: PropTypes.bool
};

Headline.defaultProps = {
  /** the headline text */
  primary: false
};

export default Headline;
