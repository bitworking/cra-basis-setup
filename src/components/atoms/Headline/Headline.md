Headline Example

    <Headline>Test</Headline>