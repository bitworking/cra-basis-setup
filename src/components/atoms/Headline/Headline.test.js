import React from 'react';
import renderer from 'react-test-renderer';
import Headline from './Headline';

it('renders snapshot', () => {
    const tree = renderer.create(
        <Headline>Test</Headline>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
