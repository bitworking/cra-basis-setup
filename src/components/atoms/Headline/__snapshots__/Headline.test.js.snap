// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Headline snapshot 1`] = `
<h1
  className="sc-bdVaJa hSyAgK"
>
  Test
</h1>
`;
